### PR TITLE
[fix](expr) Re fix BE core dump while common expr filter delete condition column

### PR DESCRIPTION
**Pull request by xinyiZzz**
## Proposed changes

Additional deleted filter condition will be materialized column be at the end of the block,
after _output_column_by_sel_idx  will be erase, we not need to filter it,
so erase it from _columns_to_filter in the first next_batch.
Eg:
      `delete from table where a = 10;`
      `select b from table;`
 a column only effective in segment iterator, the block from query engine only contain the b column,
 so no need to filter a column by expr.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -2184,21 +2175,21 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
...
-                for (const auto& i : _columns_to_filter) {
-                    ss << i << "-";
-                }
-                throw Exception(
-                        ErrorCode::INTERNAL_ERROR,
-                        "filter block column id(index) greater than block->columns(), "
-                        "column id={}, all columns that need filter={}, block columns num={}",
-                        col, ss.str().substr(0, ss.str().length() - 1), block->columns());
+        for (auto it = _columns_to_filter.begin(); it != _columns_to_filter.end();) {
+            LOG(INFO) << "_is_common_expr_column 111 " << *it;
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2188]: too many logs

Comment (xinyiZzz) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp]: fixed

**Comment by xinyiZzz** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 2]
```diff
@@ -2184,21 +2175,20 @@ Status SegmentIterator::_next_batch_internal(vectorized::Block* block) {
...
-                    ss << i << "-";
-                }
-                throw Exception(
-                        ErrorCode::INTERNAL_ERROR,
-                        "filter block column id(index) greater than block->columns(), "
-                        "column id={}, all columns that need filter={}, block columns num={}",
-                        col, ss.str().substr(0, ss.str().length() - 1), block->columns());
+        for (auto it = _columns_to_filter.begin(); it != _columns_to_filter.end();) {
+            if (*it >= block->columns()) {
+                it = _columns_to_filter.erase(it);
```
Comment (yiguolei) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:2189]: erase's cost maybe very very high?

**Review by yiguolei — DISMISSED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by gavinchou — APPROVED** [review]

**Pull request merged**